bootstrap/config/config.go

```diff
@@ -44,29 +44,55 @@ func createClient(
 	return configuration.NewConfigurationClient(providerConfig)
 }
 
-// UpdateFromProvider connects to the config provider, gets configuration, and updates the service's
+// UpdateFromProvider connects to the configuration provider, puts or gets configuration, and updates the service's
 // configuration struct.
 func UpdateFromProvider(
 	ctx context.Context,
 	startupTimer startup.Timer,
 	providerConfig configTypes.ServiceConfig,
 	serviceConfig interfaces.Configuration,
 	configStem string,
+	overwriteConfig bool,
 	lc logger.LoggingClient,
-	serviceKey string) (configuration.Client, error) {
+	serviceKey string,
+	environment *Environment,
+	overrideCount int) (configuration.Client, error) {
```

Comment:
I have some heartburn here (and some other places in the PR) where new params are being added to these methods. I would prefer extending the methods in a way that's closer to the "Open/Closed" principle. For example, the params included here might be properties on a new type (maybe called something like `ConfigOverrides`). That new type could take on properties as necessary without requiring methods that use it to take on additional params. As this functionality evolves, the number of params could become unwieldy and encapsulating related params in a type would entail fewer code changes to various method signatures involved.

Reply:
The ever growing list of function parameters indicates to me that this function is doing too much.  Trevor's suggestion would only hide the complexity and number of dependencies behind a structure that could be extended as more parameters are needed.

Decompose the function.  Alternately, wrap it and the highly cohesive ListenForChange() in a receiver and inject the necessary dependencies during construction.

Reply:
I agree that this func is trying to do too much. Specifically don't like the side effect of creating and returning the client. Will decompose.